src/azul/plugins/repository/canned/__init__.py

```diff
@@ -164,21 +167,40 @@ def count_bundles(self, source: CannedSourceRef) -> int:
             return len(staging_area.links)
         else:
             prefix = source.spec.prefix.common
-            return sum(1 for links_id in staging_area.links if links_id.startswith(prefix))
+            assert not any(map(str.isupper, prefix)), source
```

Comment:
```suggestion
            assert prefix == prefix.lower(), source
```

Given that `str.lower()` and `.__eq__` are implemented in C, this is probably an order of magnitude faster. Not that this performance difference matters terribly here, it would be a premature optimization, but I also think my version is more readable, potentially because of my background in lower-level languages. It's also shorter.